tests/integration/tests/test_util/harness/lxd.py

```diff
@@ -29,53 +29,57 @@ def __init__(self):
         self._next_id = 0
 
         self.profile = config.LXD_PROFILE_NAME
+        self.dualstack_profile = None
         self.sideload_images_dir = config.LXD_SIDELOAD_IMAGES_DIR
         self.image = config.LXD_IMAGE
         self.instances = set()
 
-        LOG.debug("Checking for LXD profile %s", self.profile)
-        try:
-            run(["lxc", "profile", "show", self.profile])
-        except subprocess.CalledProcessError:
-            try:
-                LOG.debug("Creating LXD profile %s", self.profile)
-                run(["lxc", "profile", "create", self.profile])
-
-            except subprocess.CalledProcessError as e:
-                raise HarnessError(
-                    f"Failed to create LXD profile {self.profile}"
-                ) from e
-
-        try:
-            LOG.debug("Configuring LXD profile %s", self.profile)
-            run(
-                ["lxc", "profile", "edit", self.profile],
-                input=config.LXD_PROFILE.encode(),
-            )
-        except subprocess.CalledProcessError as e:
-            raise HarnessError(f"Failed to configure LXD profile {self.profile}") from e
+        self._configure_profile(self.profile, config.LXD_PROFILE)
+
+        run(
+            [
+                "lxc",
+                "network",
+                "create",
+                config.LXD_DUALSTACK_NETWORK,
+                "ipv4.address=auto",
+                "ipv6.address=auto",
+                "ipv4.nat=true",
+                "ipv6.nat=true",
+            ],
+        )
+        self.dualstack_profile = config.LXD_DUALSTACK_PROFILE_NAME
+        self._configure_profile(
+            self.dualstack_profile,
+            config.LXD_DUALSTACK_PROFILE.replace(
+                "LXD_DUALSTACK_NETWORK", config.LXD_DUALSTACK_NETWORK
+            ),
+        )
```

Comment:
will this fail if the network exists? let's make sure it doesn't fail (follow a similar approach with what we do for the profiles)

Reply:
it will fail, refactored to match the `_configure_profile` approach